packages/tx/src/legacyTransaction.ts

```diff
@@ -106,6 +121,9 @@ export default class Transaction extends BaseTransaction<Transaction> {
 
     this.gasPrice = new BN(toBuffer(txData.gasPrice === '' ? '0x' : txData.gasPrice))
 
+    if (this.gasPrice.mul(this.gasLimit).gt(MAX_INTEGER)) {
+      throw new Error('gas limit * price overflow')
+    }
```

Comment:
Just to note that we are doing an extra multiplication with this just for the check which will cost some performance. Is this fixing a tx test or rather some VM test? If it is a VM test it might be better to do on the location where this multiplication is done anyhow? 🤔

Reply:
This was causing [this test](https://github.com/ethereum/tests/blob/develop/TransactionTests/ttGasLimit/TransactionWithGasLimitxPriceOverflow.json) to fail.